### ISSUE TITLE
in DeepAREstimator I get error prediction

### ISSUE THREAD

**Issue by ighack** [awslabs/gluon-ts#887]
```
import pandas as pd
from gluonts.dataset.common import ListDataset
from gluonts.model.deepar import DeepAREstimator
from gluonts.trainer import Trainer

df = pd.read_csv('./data/data.csv', header=0, index_col=0)

training_data = ListDataset([
    {"start": df.index[0], "target": df.value[:"2020-01-01 00:00:00"]}
], freq="1D")

estimator = DeepAREstimator(freq="1D", prediction_length=3, trainer=Trainer(epochs=10))
predictor = estimator.train(training_data=training_data)

test_data = ListDataset(
    [{"start": df.index[0], "target": df.value[:"2020-01-01 00:00:00"]}],
    freq="1D"
)


prediction = next(predictor.predict(test_data))
```

I want predict 2020-01-01,2020-01-02,2020-01-03
but I get 2019-12-29,2019-12-30,2019-12-31

data.csv

> timestamp,value
> 2019-01-01 00:00:00,NaN
> 2019-01-02 00:00:00,NaN
> 2019-01-03 00:00:00,NaN
> 2019-01-04 00:00:00,NaN
> 2019-01-05 00:00:00,NaN
> 2019-01-06 00:00:00,NaN
> 2019-01-07 00:00:00,NaN
> 2019-01-08 00:00:00,NaN
> 2019-01-09 00:00:00,NaN
> 2019-01-10 00:00:00,8000
> 2019-01-11 00:00:00,400
> …………………………
> 2019-12-17 00:00:00,7600
> 2019-12-18 00:00:00,5000
> 2019-12-19 00:00:00,13600
> 2019-12-20 00:00:00,10000
> 2019-12-21 00:00:00,NaN
> 2019-12-22 00:00:00,NaN
> 2019-12-23 00:00:00,2600
> 2019-12-24 00:00:00,10200
> 2019-12-25 00:00:00,5000
> 2019-12-26 00:00:00,800
> 2019-12-27 00:00:00,10800
> 2019-12-28 00:00:00,NaN
> 2019-12-29 00:00:00,NaN
> 2019-12-30 00:00:00,NaN
> 2019-12-31 00:00:00,3800

**Comment by lostella** [comment]
@ighack can you make your example self-contained? so that it can be copy-pasted and one can verify actual vs expected output?

**Comment by lostella** [comment]
For example, you could set "start" and "target" to some dummy values in your snippet, without involving loading from a file.

**Comment by ighack** [comment]
@ostella 

```
_data = np.random.random(365)

training_data = ListDataset([
    {"start": '2019-01-01 00:00:00', "target": _data}
], freq="1D")

estimator = DeepAREstimator(freq="1D", prediction_length=3, trainer=Trainer(epochs=10))
predictor = estimator.train(training_data=training_data)

test_data = ListDataset(
    [{"start": '2019-01-01 00:00:00', "target": _data}],
    freq="1D"
)
```
Yes,I can get correct prediction,but why.
I have replace NaN to 1234

**Comment by ighack** [comment]
I get it
I miss some data

**Comment by lostella** [comment]
@ighack the loss being optimized is negative log-likelihood, there is no harm in it being negative

**Comment by lostella** [comment]
@ighack I've adapted your example as follows:

```python
import numpy as np
from gluonts.dataset.common import ListDataset
from gluonts.model.deepar import DeepAREstimator
from gluonts.trainer import Trainer

_data = np.random.random(365)

data = ListDataset([
    {"start": '2019-01-01 00:00:00', "target": _data}
], freq="1D")

estimator = DeepAREstimator(freq="1D", prediction_length=3, trainer=Trainer(epochs=10))
predictor = estimator.train(training_data=data)

for forecast in predictor.predict(data):
    print(forecast.index)
```

and the output that gets printed is:

```
DatetimeIndex(['2020-01-01', '2020-01-02', '2020-01-03'], dtype='datetime64[ns]', freq='D')
```

So it seems to me that the right time index is associated with the forecast, do you agree?

**Comment by ighack** [comment]
@lostella 
I agree with you.

I have modify.

> timestamp,value
> 2019-01-01 00:00:00,0
> 2019-01-02 00:00:00,0
> 2019-01-03 00:00:00,0
> 2019-01-04 00:00:00,0
> 2019-01-05 00:00:00,0
> 2019-01-06 00:00:00,0
> 2019-01-07 00:00:00,0
> 2019-01-08 00:00:00,0
> 2019-01-09 00:00:00,0
> 2019-01-10 00:00:00,8000
> 2019-01-11 00:00:00,400
> ……………………
> 2019-01-27 00:00:00,0
> 2019-01-28 00:00:00,0
> 2019-01-29 00:00:00,0
> 2019-01-30 00:00:00,0
> 2019-01-31 00:00:00,0
> 2019-02-01 00:00:00,0
> 2019-02-02 00:00:00,0
> 2019-02-03 00:00:00,0
> 2019-02-04 00:00:00,0
> 2019-02-05 00:00:00,0
> 2019-02-06 00:00:00,0
> ………………
> 2019-12-05 00:00:00,2000
> 2019-12-06 00:00:00,9600
> 2019-12-07 00:00:00,0
> 2019-12-08 00:00:00,0
> 2019-12-09 00:00:00,0
> 2019-12-10 00:00:00,12800
> 2019-12-11 00:00:00,23600
> 2019-12-12 00:00:00,3400
> 2019-12-13 00:00:00,10800
> 2019-12-14 00:00:00,0
> 2019-12-15 00:00:00,0
> 2019-12-16 00:00:00,0
> 2019-12-17 00:00:00,7600
> 2019-12-18 00:00:00,5000
> 2019-12-19 00:00:00,13600
> 2019-12-20 00:00:00,10000
> 2019-12-21 00:00:00,0
> 2019-12-22 00:00:00,0
> 2019-12-23 00:00:00,2600
> 2019-12-24 00:00:00,10200
> 2019-12-25 00:00:00,5000
> 2019-12-26 00:00:00,800
> 2019-12-27 00:00:00,10800
> 2019-12-28 00:00:00,0
> 2019-12-29 00:00:00,0
> 2019-12-30 00:00:00,0
> 2019-12-31 00:00:00,3800

I also get get correct prediction

(Timestamp('2020-01-01 00:00:00', freq='D'), -7453.587890625)
(Timestamp('2020-01-02 00:00:00', freq='D'), -1752.99072265625)
(Timestamp('2020-01-03 00:00:00', freq='D'), -2313.981201171875)

but I get minus value

**Comment by lostella** [comment]
@ighack sorry, I misunderstood what you meant by negative values.

I think substituting NaNs with zeros is not the right thing to do, as the model will take those zeros as genuine data points instead of recognizing them as missing. Could you try keeping them in the data? For example, as np.nan values

**Comment by ighack** [comment]
@lostella 

zero means we don't sell any goods,so I use zero replace NaN.

I don't  think sale is minus value.
that is incorrect.

**Comment by lostella** [comment]
@ighack I understand, thanks for the explanation. Then I think the default output distribution (student's t distribution) is not ideal. You may want to try out something for positive count data, like negative binomial:

```python
from gluonts.distribution import NegativeBinomialOutput
from gluonts.model.deepar import DeepAREstimator
from gluonts.trainer import Trainer

estimator = DeepAREstimator(
    freq="1D",
    prediction_length=3,
    distr_output=NegativeBinomialOutput(),
    trainer=Trainer(epochs=10)
)
```

**Comment by ighack** [comment]
@lostella 
Thank for you help, I get correct result.

**Comment by lostella** [comment]
@ighack cool! Closing the issue, but feel free to post further questions you may have on the matter
